Packet++/header/TcpReassembly.h

```diff
@@ -197,16 +205,26 @@ struct TcpReassemblyConfiguration
 	 */
 	uint32_t maxOutOfOrderFragments;
 
+	/**  To enable to clear buffer once packet contains data from a different side than the side seen before
+	 */
+	bool enableBaseBufferClearCondtion;
+
 	/**
 	 * A c'tor for this struct
 	 * @param[in] removeConnInfo The flag indicating whether to remove the connection data after a connection is closed. The default is true
 	 * @param[in] closedConnectionDelay How long the closed connections will not be cleaned up. The value is expressed in seconds. If it's set to 0 the default value will be used. The default is 5.
 	 * @param[in] maxNumToClean The maximum number of items to be cleaned up per one call of purgeClosedConnections. If it's set to 0 the default value will be used. The default is 30.
 	 * @param[in] maxOutOfOrderFragments The maximum number of unmatched fragments to keep per flow before missed fragments are considered lost. The default is unlimited.
+	 * @param[in] enableBaseBufferClearCondtion To enable to clear buffer once packet contains data from a different side than the side seen before
 	 */
-	TcpReassemblyConfiguration(bool removeConnInfo = true, uint32_t closedConnectionDelay = 5, uint32_t maxNumToClean = 30, uint32_t maxOutOfOrderFragments = 0) :
-		removeConnInfo(removeConnInfo), closedConnectionDelay(closedConnectionDelay), maxNumToClean(maxNumToClean), maxOutOfOrderFragments(maxOutOfOrderFragments)
+	TcpReassemblyConfiguration(bool removeConnInfo = true, uint32_t closedConnectionDelay = 5, uint32_t maxNumToClean = 30, uint32_t maxOutOfOrderFragments = 0, 
+		bool enableBaseBufferClearCondtion = true) : removeConnInfo(removeConnInfo), closedConnectionDelay(closedConnectionDelay), maxNumToClean(maxNumToClean), maxOutOfOrderFragments(maxOutOfOrderFragments), enableBaseBufferClearCondtion(enableBaseBufferClearCondtion)
 	{
+		if(!enableBaseBufferClearCondtion && maxOutOfOrderFragments==0)
+		{
+			// set a default value to prevent indefinate filling of out of order buffer 
+			maxOutOfOrderFragments = 20;

```

Comment:
I'm not sure we need a default value here. If users set the value of` `maxOutOfOrderFragments` to be `0` they expect the buffer to never be cleaned. Setting a value for them is kind of weird behavior. WDYT?

Reply:
Yeah setting a default value will be bad according to previous functionality of maxOutOfOrderFrafments. I have removed this